tests/unit_tests/commandline/cli_top_level_tester.py

```diff
@@ -76,6 +77,59 @@ def test_datasets(self):
             ]
         return self._test_datasets
 
+    def retrieve_selected_columns_from_list_command(self, args):
+        """If --columns was used for a 'list' command, retrieve the list of columns.
```

Comment:
Is this only used for "list" commands? If so, should this be moved to tests relating to the "list" command or do we anticipate having other CLI commands that will output columns? If so, should this be renamed to just be something like "retrieve_table_columns"?

Reply:
I could foresee that this might be used by the ``get / describe`` command in the future. Currently it's only list commands, but if we wanted to get a plugin, only retrieving information about its ``package, family, and docstring``, then we could apply this to those commands as well.

Reply:
If we foresee using this for other commands, let's rename this to just `retrieve_selected_columns` and update the docstring to remove references to the `list` command. 

If we only foresee using this for `list`, let's move it into the tests relating to `list`.

Reply:
Done!